staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/controller.go

```diff
@@ -292,6 +293,35 @@ func (c *celAdmissionController) Validate(
 					continue
 				}
 			}
+
+			// Ensure param is populated with its GVK for consistency
+			// (CRD dynamic informer always returns objects with kind/apiversion,
+			// but native types do not include populated TypeMeta.
+			if param != nil {
+				if paramGVK := param.GetObjectKind(); paramGVK.GroupVersionKind().Empty() {
+					// Very unfortunate. Unfortunately object is shared and cannot
+					// modify GVK async
+					param = param.DeepCopyObject()
```

Comment:
this deep copy is very unfortunate. It would be better if we could do a shallow copy somehow to be able to set the top-level GVK.

Any ideas?

Reply:
populate its GVK in a `sync.Once`?

Reply:
I think I've been able to work around this by using `SharedInformer`'s `SetTransform` to set GVK before it goes into the `Store`.

It is still unfortunate though that this just works around `WithoutVersionDecoder`'s wiping it out, but I do not see a way with the current SDK to look up a SharedInformer dynamically by its GVK and be able to customize the `NegotiatedSerializer` to not remove GVK.